src/components/layout/footer/newsletter-box.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+import { useFormik, FormikProps } from 'formik'
+import * as Yup from 'yup'
+import * as S from './styles'
+
+export interface NewsletterFormValues {
+  email: string
+}
+
+export interface NewsletterConfig {
+  onSubmit: (values: NewsletterFormValues) => void
+}
+
+export interface NewsletterProps {
+  values: NewsletterFormValues
+  errors: FormikProps<NewsletterFormValues>['errors']
+  touched: FormikProps<NewsletterFormValues>['touched']
+  handleSubmit: FormikProps<NewsletterFormValues>['handleSubmit']
+  handleChange: FormikProps<NewsletterFormValues>['handleChange']
+  handleBlur: FormikProps<NewsletterFormValues>['handleBlur']
+}
+
+export const useNewsletterForm = ({
+  onSubmit,
+}: NewsletterConfig): NewsletterProps => {
+  const fakeErrorMessage = 'fake'
+  const formik = useFormik({
+    initialValues: {
+      email: '',
+    },
+    validationSchema: Yup.object({
+      email: Yup.string().email(fakeErrorMessage).required(fakeErrorMessage),
+    }),
+    onSubmit: onSubmit,
+  })
+
+  return {
+    values: formik.values,
+    errors: formik.errors,
+    touched: formik.touched,
+    handleSubmit: formik.handleSubmit,
+    handleChange: formik.handleChange,
+    handleBlur: formik.handleBlur,
+  }
+}
+
+export const onSubmitNewsletterForm = (values: NewsletterFormValues): void => {
+  alert(JSON.stringify(values, null, 2))
+}
+
+const Newsletter: React.FC = () => {
+  const t = {
+    headings: {
+      newsletter: 'Newsletter',
+    },
+    newsletter: {
+      note:
+        'Chcete vědět, na čem pracujeme? Jednou za měsíc shrneme, co se v komunitě událo a co chystáme.',
+      inputPlaceholder: 'Zadejte e-mail',
+      inputErr: 'Zadejte prosím validní e-mailovou adresu.',
+      subscribe: 'Odebírat',
+    },
+  }
+
+  const form = useNewsletterForm({
+    onSubmit: onSubmitNewsletterForm,
+  })
+
+  return (
+    <S.Newsletter>
+      <S.NewsletterIcon />
+      <S.Heading>{t.headings.newsletter}</S.Heading>
+      <S.NewsletterInfo>{t.newsletter.note}</S.NewsletterInfo>
+      <S.NewsletterForm onSubmit={form.handleSubmit}>
+        <S.NewsletterFormControl>
+          <S.NewsletterInput
+            name="email"
+            placeholder={t.newsletter.inputPlaceholder}
+            onChange={form.handleChange}
+            onBlur={form.handleBlur}
+            value={form.values.email}
+          />
+        </S.NewsletterFormControl>
+        <S.NewsletterButton>{t.newsletter.subscribe}</S.NewsletterButton>
+        <S.NewsletterInputErrMessage
+          className={
+            form.touched.email && form.errors?.email ? 'is-visible' : ''
```

Comment:
V tomto případě je vhodnější použít buď přímo styled components props anebo pak: `{form.touched.email && form.errors?.email && <S.Newslet...`. 

V samotném elementu je pak lepší využít přímo Formik error (chybové hlášky tedy mít na vstupu `useNewsletterForm` kvůli překladům).

Reply:
Jo, tohle je úplně debilní, jak jsem psal výše. Je to kvůli tomu, že na té třídě je pověšená CSS transition (opacity). Problém je, že když se využije formik error, tak ve chvíli kdy zmizí, tak zmizí i text a animace se "usekne", protože není co animovat. Tohle právě nevím, jak se v Reactu standardně řeší. Dokud se tohle nerozsekne, tak nevím jak dál (případně samozřejmě zrušit tu transition je též řešení 😆

Reply:
Zrušení transition je úplně OK 😉

Reply:
Nakonec jsem použil styled components props, původně jsem tam měl tu variantu s `{form.touched.email && form.errors?.email && <S.Newslet...`. Změnil jsem to, protože jsem si nevěděl rady s tím, že když se poprvé zobrazí ten error, tak se vždycky cukne text na celé stránce. Nechápu proč a ani nevím, co s tím.